src/Fixer/ClassNotation/MethodSeparationFixer.php

```diff
@@ -76,159 +113,187 @@ public function getPriority()
+            if ('method' === $element['type'] && !$tokens[$class]->isGivenKind(T_INTERFACE)) {
+                // method of class or trait
+                $attributes = $tokensAnalyzer->getMethodAttributes($index);
 
-            if ($tokens[$index]->isGivenKind(T_INTERFACE)) {
-                $this->fixInterface($tokens, $classStart, $classEnd);
-            } else {
-                // classes and traits can be fixed the same way
-                $this->fixClass($tokens, $tokensAnalyzer, $classStart, $classEnd);
+                $methodEnd = true === $attributes['abstract']
+                    ? $tokens->getNextTokenOfKind($index, [';'])
+                    : $tokens->findBlockEnd(Tokens::BLOCK_TYPE_CURLY_BRACE, $tokens->getNextTokenOfKind($index, ['{']))
+                ;
+
+                $this->fixSpaceBelowClassMethod($tokens, $classEnd, $methodEnd);
+                $this->fixSpaceAboveClassElement($tokens, $classStart, $index);
+
+                continue;
             }
+
+            // `const`, `property` or `method` of an `interface`
+            $this->fixSpaceBelowClassElement($tokens, $classEnd, $tokens->getNextTokenOfKind($index, [';']));
+            $this->fixSpaceAboveClassElement($tokens, $classStart, $index);
         }
     }
 
     /**
-     * @param Tokens         $tokens
-     * @param TokensAnalyzer $tokensAnalyzer
-     * @param int            $classStart
-     * @param int            $classEnd
+     * {@inheritdoc}
      */
-    private function fixClass(Tokens $tokens, TokensAnalyzer $tokensAnalyzer, $classStart, $classEnd)
+    protected function createConfigurationDefinition()
     {
-        for ($index = $classEnd; $index > $classStart; --$index) {
-            if (!$tokens[$index]->isGivenKind(T_FUNCTION) || $tokensAnalyzer->isLambda($index)) {
-                continue;
-            }
-
-            $attributes = $tokensAnalyzer->getMethodAttributes($index);
-            if (true === $attributes['abstract']) {
-                $methodEnd = $tokens->getNextTokenOfKind($index, [';']);
-            } else {
-                $methodStart = $tokens->getNextTokenOfKind($index, ['{']);
-                $methodEnd = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_CURLY_BRACE, $methodStart, true);
-            }
-
-            $this->fixSpaceBelowMethod($tokens, $classEnd, $methodEnd);
-            $this->fixSpaceAboveMethod($tokens, $classStart, $index);
-        }
+        $types = ['const', 'method', 'property'];
+
+        return new FixerConfigurationResolver([
+            (new FixerOptionBuilder('elements', sprintf('List of classy elements; \'%s\'.', implode("', '", $types))))
+                ->setAllowedTypes(['array'])
+                ->setAllowedValues([
+                    (new FixerOptionValidatorGenerator())->allowedValueIsSubsetOf($types),
+                ])
+                ->setDefault(['method']) // TODO Rename the MethodSeparationFixer in 3.0. Set default to ['const', 'method', 'property']
```

Comment:
I would make it already on 2.x line to decrease amount of work needed to migrate 2.x->3.x
that way, on 2.x we will have future-compat layer already.

rename the fixer already and then create `MethodSeparationFixer` once more that will be only a proxy fixer with unmodified behaviour, no configuration options, and will be deprecated.